src/node_api.cc

```diff
@@ -920,13 +920,13 @@ NAPI_NO_RETURN void napi_fatal_error(const char* location,
                                      size_t message_len) {
   char* location_string = const_cast<char*>(location);
   char* message_string = const_cast<char*>(message);
-  if (location_len != -1) {
+  if (static_cast<int>(location_len) != -1) {
```

Comment:
size_t might allow for larger values than signed ints and we're wrongly casting `2^n-1` to `-1` (where `n` is how big size_t is). If we want to allow `-1` here, why don't we change the parameter type? Or could we use `0` instead of `-1` to signal that we don't want to set the `message_string`?